File: src/main/java/com/optica/services/login/SecurityService.java
package com.optica.services.login;

import javax.servlet.http.HttpServletRequest;

import org.springframework.security.core.userdetails.UserDetails;

import com.optica.entity.User;

public interface SecurityService {

	String findLoggedInUsername();

	void autologin(String username, String password, HttpServletRequest request);

	User getCurrentUser();

}

File: src/main/kotlin/com/optica/services/login/SecurityService.kt
package com.optica.services.login

import javax.servlet.http.HttpServletRequest

import org.springframework.security.core.userdetails.UserDetails

import com.optica.entity.User

interface SecurityService {

    val currentUser: User

    fun findLoggedInUsername(): String

    fun autologin(username: String?, password: String?, request: HttpServletRequest)

}
